test/bntest.c

```diff
@@ -82,6 +82,7 @@
 #include <openssl/err.h>
 
 #include "../crypto/bn/bn_lcl.h"
+#include "../crypto/bn/bn_prime.h"
```

Comment:
Both these lines are problematic. It probably seems fine... until you configure with "shared" on Linux, Windows or VMS, where there's a tight control on what symbols get exported (the symbols in those header files are not). 

I haven't looked deeply enough and won't be able today... But it seems to me that the tests you add are unit tests rather than overall published function tests.